apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/properties/datastore/configuration/DataStorePropertiesConfiguration.java

```diff
@@ -48,25 +51,65 @@ DataStoreProperties configuration(
 
     // if we should not read from the bridge, go for defaults
     if (dataStoreConfig.ignoreBridge()) {
+      LOG.info("DataStoreConfig.ignoreBridge() == true, will use pre-configured defaults");
       return fromConfig;
     }
 
+    LOG.info(
+        "DataStoreConfig.ignoreBridge() == false, will try to fetch data store metadata using the Bridge");
+    final AtomicInteger callCount = new AtomicInteger(0);
     try {
       // fire request
-      Schema.SupportedFeaturesRequest request =
-          Schema.SupportedFeaturesRequest.newBuilder().build();
-      Schema.SupportedFeaturesResponse supportedFeatures = bridge.getSupportedFeatures(request);
+      Schema.SupportedFeaturesResponse supportedFeatures =
+          fetchSupportedFeatures(bridge, callCount);
 
       // construct props from bridge
-      return new DataStorePropertiesImpl(
-          supportedFeatures.getSecondaryIndexes(),
-          supportedFeatures.getSai(),
-          supportedFeatures.getLoggedBatches());
+      DataStorePropertiesImpl props =
+          new DataStorePropertiesImpl(
+              supportedFeatures.getSecondaryIndexes(),
+              supportedFeatures.getSai(),
+              supportedFeatures.getLoggedBatches());
+      LOG.info("Successfully fetched data store metadata ({} retries)", callCount.get() - 1);
+      return props;
     } catch (Exception e) {
-      LOG.warn(
-          "Error fetching the data store properties from the bridge, fallback to the configuration based properties.",
-          e);
-      return fromConfig;
+      final String msgBase =
+          "Failed to fetch data store metadata ({} retries)".formatted(callCount.get() - 1);
+      if (dataStoreConfig.bridgeFallbackEnabled()) {
+        LOG.warn(
+            msgBase + ", 'bridgeFallbackEnabled' == true, will return fallback data store metadata",
+            e);
+        return fromConfig;
+      }
+      LOG.warn(msgBase + ", 'bridgeFallbackEnabled' == false, will fail");
+      throw e;
+    }
+  }
+
+  /**
+   * Method for fetching data store metadata: will attempt up to 9 retries, for up to 1 minute
+   * before failing.
+   */
+  @Retry(
+      maxRetries = 9,
```

Comment:
is `9` really needed? I mean even `5` seems too high to me..

Reply:
Round number like 10,  but mostly that it'd try for closer to 1 minute, but without super long delay in-between.
We can change it of course. Depends on what kind of problem/outage we'd have.

Reply:
I'll change this back to 5 total calls (4 retries); can tune later on if need be.